src/compose/service.ts

```diff
@@ -600,7 +600,7 @@ export class Service {
 				'Attempt to build Service class from container with malformed labels',
 			);
 		}
-		const nameMatch = container.Name.match(/.*_(\d+)_(\d+)$/);
+		const nameMatch = container.Name.match(/.*_(\d+)_(\d+)(?:_(.*?))?$/);
 		if (nameMatch == null) {
 			throw new InternalInconsistencyError(
 				'Attempt to build Service class from container with malformed name',
```

Comment:
We can start adapting the Go-style error reporting pattern by adding expected & actual container name to this error message

Reply:
That's a good idea, although I'd only expect to see this error if you try running an older supervisor with the new container names